Services/Notifications/classes/Setup/ilNotificationUpdateSteps.php

```diff
@@ -168,4 +168,16 @@ public function step_9(): void
         );
 
     }
+
+    public function step_10(): void
+    {
+        if (!$this->db->tableColumnExists('notification_osd', 'identification')) {
+            $this->db->addTableColumn('notification_osd', 'identification', [
```

Comment:
What about existing entries? I see two options:

1) Truncate `notification_osd`
2) Using the internal ID as `identification`

Why? Because an empty ID can be an issue when passing it to the `GlobalScreen` or the `UI Toasts`, or even when handling the `callbacks` (maybe needs some extra handling for empty strings)`.

Reply:
I would favor to not set a default just for the purpose to have it, but i would not truncate the table without the adminstrators awareness, since it may contain old but upcoming notifications.
Atm the table will be truncated when the notifications are disabled. Else the empty identifications should cause any trouble. The global screen uses (and will further use) the real id of the notifications and for the UI Toast the setting of the default would be as inconvenient as the empty one.
And as far as i planned the callbacks should anyways be that strict to only handle only their own created notifications and not past ones wich would defy their limited scoping too. Therefore i would approve the disabling of the setting to be the only case in wich "lost" notifications can be deleted.

Therefore i dont see a need for a default here.